### PR TITLE
feat(rust, python): introduce 'label' instead of 'truncate' in group_by_dynamic, which can take `label='right'`

### LINKED ISSUE #9134: add label='right' option to group_by_dynamic
### Polars version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Issue description

When passing `-every` and `+every`  to the offset parameter polars returns different results.
My understanding is that these should be equivalent because the aggregation grid shifted by `n*every` for all integer `n` is equal.

### Reproducible example

```python
X = pl.DataFrame(dict(
    time = ["2023-05-30 08:01:00", "2023-05-30 08:23:00", "2023-05-30 09:13:00"],
    value = [1, 2, 3],
)).with_columns(
    pl.col("time").str.strptime(pl.Datetime),
).set_sorted("time")


# 1. Gives times [08:00, 09:00]:
X.groupby_dynamic("time", every="30m", offset="-30m").agg(pl.col("value").sum())

# 2. Gives times [07:59, 08:59]:
X.groupby_dynamic("time", every="30m", offset="-1m").agg(pl.col("value").sum())

# 3. Gives times [08:00, 09:00]:
X.groupby_dynamic("time", every="30m", offset="0m").agg(pl.col("value").sum())

# 4. Gives times [08:01, 09:01]:
X.groupby_dynamic("time", every="30m", offset="1m").agg(pl.col("value").sum())

# 5. Gives times [09:00]:
X.groupby_dynamic("time", every="30m", offset="30m").agg(pl.col("value").sum())
```

### Expected behavior

Examples 1., 3. and 5. should give identical results.

### Installed versions

<details>

```
---Version info---
Polars: 0.17.11
Index type: UInt32
Platform: Linux-4.18.0-372.13.1.el8_6.x86_64-x86_64-with-glibc2.28
Python: 3.10.10 | packaged by conda-forge | (main, Mar 24 2023, 20:08:06) [GCC 11.3.0]
---Optional dependencies---
numpy: 1.24.3
pandas: 1.5.3
pyarrow: 11.0.0
connectorx: <not installed>
deltalake: <not installed>
fsspec: 2023.5.0
matplotlib: 3.7.1
xlsx2csv: <not installed>
xlsxwriter: <not installed>
```

</details>

### PR DESCRIPTION
**Pull request by MarcoGorelli**
closes #9134 

not correct (yet), but I'm out of time for today 🛌

### CODE REVIEW

File: py-polars/tests/unit/operations/rolling/test_rolling.py [hunk 1]
```diff
@@ -554,7 +554,7 @@ def test_group_by_dynamic_startby_5599(tzinfo: ZoneInfo | None) -> None:
         "date",
         every="31m",
         include_boundaries=True,
-        truncate=False,
+        label="datapoint",
         start_by="datapoint",
```
Comment (MarcoGorelli) [inline py-polars/tests/unit/operations/rolling/test_rolling.py:558]: separate to this PR, but I don't understand why there's so many `group_by_dynamic` tests in this file

I think they need their own file

for now I'm adding them here, but then I'd like to do a separate "chore" PR of pulling them out into their own file

**Comment by ritchie46** [comment]
Nice! Maybe we can include this in the release :eyes:

File: py-polars/tests/unit/operations/rolling/test_rolling.py (hunk 1 above)
Comment (ritchie46) [reply py-polars/tests/unit/operations/rolling/test_rolling.py:558]: :+1: If we can move them to `polars/it` that would be even better. (might not be possible). So either way is fine.

**Review by ritchie46 — APPROVED** [review]
Great stuff @MarcoGorelli! This will be a good release for time-series. :)

**Pull request merged**